File: Algorithm/10828_스택/10828_스택.cpp
#include <iostream>
#include <stack>
#include <string>

using namespace std;

int main(void)
{
    int N;
    cin >> N;

    stack<int> _stack;
    string op;

    for (int i = 0; i < N; i++)
    {
        cin >> op;
        if (op == "push")
        {
            int num = 0;
            cin >> num;
            _stack.push(num);
        }
        if (op == "pop")
        {
            if (!_stack.empty())
            {
                cout << _stack.top() << endl;
                _stack.pop();
            }
            else
                cout << -1 << endl;
        }
        if (op == "size")
            cout << _stack.size() << endl;
        if (op == "empty")
        {
            if (_stack.empty())
                cout << 1 << endl;
            else
                cout << 0 << endl;
        }
        if (op == "top")
        {
            if (_stack.empty())
                cout << -1 << endl;
            else
                cout << _stack.top() << endl;
        }
    }
}

File: Algorithm/10828_스택/10828_스택.py
import sys

N = int(input())
stack = []

for _ in range(N):
    # split() : 공백 N개도 1개로 처리
    #             탭이나 엔터도 처리
    # split(' ') : 공백 1개만 처리
    op = sys.stdin.readline().split()

    if op[0] == 'push':
        stack.append(int(op[1]))

    elif op[0] == 'pop': 
        # stack.count(val) : val이 리스트 안에 몇개인지
        # len(stack) : 리스트 길이가 얼마인지
        if len(stack) > 0:
            print(stack.pop())
        else:
            print(-1)

    elif op[0] == 'size':
        print(len(stack))

    elif op[0] == 'empty':
        if len(stack) > 0:
            print(0)
        else:
            print(1)

    elif op[0] == 'top':
        if len(stack) > 0:
            print(stack[-1])
        else:
            print(-1)
